itests/util/src/main/java/org/apache/hadoop/hive/ql/externalDB/AbstractExternalDB.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.externalDB;
+
+import sqlline.SqlLine;
+
+import org.apache.commons.io.output.NullOutputStream;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileReader;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.IOException;
+import java.net.URI;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * abstractExternalDB is incharge of connect to and populate externaal database for qtest
+ */
+public abstract class AbstractExternalDB {
+    protected static final Logger LOG = LoggerFactory.getLogger("AbstractExternalDB");
+
+    protected static final String userName = "qtestuser";
+    protected static final String password = "qtestpassword";
+    protected static final String dbName = "qtestDB";
+
+    public String externalDBType = "derby"; // default: derby
+    protected String url = String.format("jdbc:derby:memory:%s;create=true", dbName); // defualt: dervy
```

Comment:
nit: Fixed the typo. default: derby

Reply:
Could you fix "defualt: dervy" ?

Reply:
Good catch. Will change

Reply:
On second thought, this driver does not support derby currently. (The code is there but haven't really tested it). Maybe I should change it to the default to mysql.